python-checks/src/main/java/org/sonar/python/checks/EmptyNestedBlockCheck.java

```diff
@@ -52,11 +60,19 @@ public void initialize(Context context) {
       // sublist call is excluding last index and token following last token of statement list (dedent) should be included in the comment verification.
       int to = parentTokens.indexOf(statementListTree.lastToken()) + 2;
       if (!containsComment(parentTokens.subList(from, to))) {
-        if (statementListTree.statements().isEmpty()) {
-          ctx.addIssue(statementListTree.firstToken(), MESSAGE);
-        } else {
-          ctx.addIssue(statementListTree.statements().get(0), MESSAGE);
-        }
+        var passTreeElement = Optional.of(statementListTree)
+          .map(StatementList::statements)
+          .map(Collection::stream)
+          .flatMap(Stream::findFirst)
+          .map(Tree.class::cast)
+          .orElseGet(statementListTree::firstToken);
+
+        var quickFix = PythonQuickFix.newQuickFix(QUICK_FIX_MESSAGE)
+          .addTextEdit(PythonTextEdit.insertLineBefore(passTreeElement, TODO_COMMENT_TEXT))
```

Comment:
I think there is one problem here if the `pass` statement is on the same line as the parent block.
```
def foo():
  for i in range(3): pass
```

Currently it seems to produce something like (looking at the unit tests - haven't confirmed in an actual IDE):

```
def foo():
  for i in range(3): # TODO: Add implementation
                              pass
```

So maybe there is a problem with the method `insertLineBefore` as well, because I would have instinctively expected it to produce:

```
def foo():
  # TODO: Add implementation
  for i in range(3): pass
```
(Note that this wouldn't work either, as I think this snippet would also raise an issue)


Ideally we probably want to produce:

```
def foo():
  for i in range(3): 
    # TODO: Add implementation
    pass
```

And it may be okay to have a separate ticket for this.

Reply:
A separate issue created SONARPY-1277